tests/model_fields/test_jsonfield.py

```diff
@@ -636,6 +636,43 @@ def test_has_key_number(self):
                     [obj],
                 )
 
+    def test_has_key_special_chars(self):
+        value = {
+            "single'": [],
+            'double"': [
+                connection.features.supports_double_quotes_in_json_field_lookups,
+            ],
+            "dollar$": [],
+            "dot.dot": [],
+            "with space": [],
+            "back\\slash": [],
+            "question?mark": [],
+            "user@name": [],
+            "emo🤡'ji": [],
+            "com,ma": [],
+            "curly{{{brace}}}s": [],
+            "escape\uffff'sequence": [],
+            "m\\i@x. m🤡'a,t{{{ch}}}e?d$\"'es\uffff'cape": [
```

Comment:
I'm actually surprised that these two work, because I would've expected the `\uffff'` would escape the path expression and trigger the issue.

Reply:
I'm surprised as well, I'll get the Oracle suite running to see what it has to say.

Reply:
I’ve updated the tests to include has_keys, has_any_keys, and path exact lookups.

On the escaping, my hunch is that Oracle looks for the last occurrence of the delimiter, so these cases do not trigger the issue.

Reply:
Oh, actually @charettes, I think it's because we use `json.dumps()` for the keys!

```pycon
>>> print("\uffff")
￿
>>> print(json.dumps("\uffff"))
"\uffff"
>>> print("hello\uffff'world")
hello￿'world
>>> print(json.dumps("hello\uffff'world"))
"hello\uffff'world"
```

So, what happens is: in the SQL template, we use the `\uffff` character as-is (`￿`). However, in the keys, they get escaped into `\uffff`, which is also a valid representation for `￿` in JSON. In JS:

```js
JSON.parse(`"hello\uffff'world"`)
// returns "hello￿'world"
```

Hence, Oracle does not consider the `\uffff'` to be the same as the delimiter. I think it _should_ be bulletproof then!